app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,20 @@
+class SessionsController < ApplicationController
+  def new
+  end
+
+  def create
+    user = User.find_by email: params[:session][:email].downcase
+    if user && user.authenticate(params[:session][:password])
+      log_in user
+      redirect_to user
+    else
+      flash.now[:danger] = t "session_danger"
```

Comment:
Theo mình hiểu trong mục 5.1: http://guides.rubyonrails.org/action_controller_overview.html#the-flash 
nếu k dùng flash.now mà dùng flash  thông thường thì trong trường hợp user login failed nếu user gửi thêm 1 request login nũa thì thông báo đó vẫn hiện ra.